github-pullrequest-plugin/src/main/java/com/github/kostyasha/github/integration/branch/events/impl/GitHubBranchCommitEvents.java

```diff
@@ -0,0 +1,147 @@
+package com.github.kostyasha.github.integration.branch.events.impl;
+
+import com.github.kostyasha.github.integration.branch.GitHubBranch;
+import com.github.kostyasha.github.integration.branch.GitHubBranchCause;
+import com.github.kostyasha.github.integration.branch.GitHubBranchRepository;
+import com.github.kostyasha.github.integration.branch.GitHubBranchTrigger;
+import com.github.kostyasha.github.integration.branch.events.GitHubBranchCommitEvent;
+import com.github.kostyasha.github.integration.branch.events.GitHubBranchCommitEventDescriptor;
+import com.github.kostyasha.github.integration.branch.events.GitHubBranchEvent;
+import com.github.kostyasha.github.integration.branch.events.GitHubBranchEventDescriptor;
+
+import hudson.Extension;
+import hudson.model.TaskListener;
+
+import net.sf.json.JSONObject;
+
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.github.GHBranch;
+import org.kohsuke.github.GHCompare;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.StaplerRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.CheckForNull;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * @author Kanstantsin Shautsou
+ */
+public class GitHubBranchCommitEvents extends GitHubBranchEvent {
+    private static final String DISPLAY_NAME = "Commit Events";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GitHubBranchCommitEvents.class);
+
+    private List<GitHubBranchCommitEvent> events = new ArrayList<>();
```

Comment:
Even in events sounds weird. initial termine Event was tied to events coming from GH hooks, but as soon as trigger started working with complex cases that not really events. That should be called as check/trigger cases or somehow else.